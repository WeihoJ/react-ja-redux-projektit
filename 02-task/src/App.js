import './App.css';
import Person from './person/person.js';

const personel = [
  { name: "Mio", age: 17},
  { name: "Jaska", age: 18},
  { name: "Jorma", age: 19}
]


const App = () => {
  return (
    <div className="App">
      <Person name={personel[0].name} age={personel[0].age} />
      <Person name={personel[1].name} age={personel[1].age} />
      <Person name={personel[2].name} age={personel[2].age} />
    </div>
  );
}

export default App;
