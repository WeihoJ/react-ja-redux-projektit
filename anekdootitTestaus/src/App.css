body {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
}

* {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Helvetica Neue', monospace;
}

.App {
    position: absolute;
    top: 50px;
    left: 50%;
    transform: translateX(-50%);
    display: flex;
    justify-content: center;
    align-items: center;
    max-width: 800px;
}

.anecdote p {
    color: #F2E9CD;
    font-size: 18px;
    text-align: left;
}

.anecdote {
    background-color: #296118;
    padding: 5px 10px;
    border-radius: 5px;
    margin-bottom: 10px;
    width: 800px;
}

.votes {
    padding: 15px;
}

.napit {
    display: flex;
    align-items: center;
    justify-content: space-evenly;
}

/* .anecdoteBtn,
.voteBtn{
    background-color: #983430;
    color: #F2E9CD;
    border: none;
    width: 50px;
    height: 25px;
} */

button {
    background-color: #983430;
    color: #F2E9CD;
    border: none;
    width: 200px;
    height: 50px;
    border-radius: 10px;
    font-size: 15px;
}

.anecdoteVotes span {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 0;
    padding: 5px;
}

.anecdoteVotes {
    font-size: 18px;
}

.votes :first-child {
    background-color: #3B898A;
    color: #DEF3F3;
}

.votes :last-child {
    background-color: #000000;
    color: #DDDB49;
}

.strong {
    font-weight: bold;
    font-size: 25px;
}

.medium {
    font-size: 20px;
}