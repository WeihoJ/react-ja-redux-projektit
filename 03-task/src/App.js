import React from 'react';
import './App.css'
import Laskuri from './Laskuri'

const App = () => {
  return (
    <Laskuri />
  );
}


export default App;