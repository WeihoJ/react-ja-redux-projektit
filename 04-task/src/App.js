import './App.css';
import Form from './components/Form';
import View from './components/View';


const App = () => {
  return (
    <div className="App">
      <Form />
    </div>
  );
};

export default App;