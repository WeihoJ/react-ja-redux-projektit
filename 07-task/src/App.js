import './App.css';
import Syote from './komponentit/Syote';

function App() {
  return (
    <div className="keskitys">
      <Syote />
    </div>
  );
}

export default App;
