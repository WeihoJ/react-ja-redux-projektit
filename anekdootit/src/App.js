import React, { useState } from "react";
import './App.css';
import Anecdote from "./components/Anecdote";

const App = () => {

  return (
    <div className="App">
      <Anecdote />
    </div>
  );
}

export default App;