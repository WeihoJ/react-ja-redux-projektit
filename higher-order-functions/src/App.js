import './App.css';
import React from 'react';

function App() {
  const animals = [
    { name: 'Fluffykins', species: 'rabbit' },
    { name: 'Caro', species: 'dog' },
    { name: 'Hamilton', species: 'dog' },
    { name: 'Harold', species: 'fish' },
    { name: 'Ursula', species: 'cat' },
    { name: 'Jimmy', species: 'fish' }
  ];

  // console.log(animals)

  // let dogs = []
  // for (let i = 0; i < animals.length; i++) {
  //   if (animals[i].species === 'dog') {
  //     dogs.push(animals[i])
  //   }
  // }

  // console.log(dogs)


  // const result = animals.filter(animal => {
  //   return animal.species === 'dog'
  // });

  // console.log(result)

  const otherAnimals = animals.filter(animal => {
    return animal.species !== 'dog'
  });

  console.log(otherAnimals)



  return (
    <div className="App">

    </div>
  );
}

export default App;
