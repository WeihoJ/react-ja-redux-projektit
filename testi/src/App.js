import './App.css';

const App = () => {
    return (
        <div className="App">
            <h1>Testing testing ...</h1>
        </div>
    );
};

export default App;
