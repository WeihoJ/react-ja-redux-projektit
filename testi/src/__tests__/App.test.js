import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

it('pitäis tehdä snapshot', () => {
    const { asFragment } = render(<App />);

    expect(asFragment(<App />)).toMatchSnapshot();
});
