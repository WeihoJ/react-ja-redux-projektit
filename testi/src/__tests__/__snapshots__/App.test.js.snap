// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pitäis tehdä snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Testing testing ...
    </h1>
  </div>
</DocumentFragment>
`;
